hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -202,11 +203,18 @@ private List<String> getPartitionPathsForIncrementalCleaning(HoodieCleanMetadata
   /**
    * Scan and list all partitions for cleaning.
    * @return all partitions paths for the dataset.
-   * @throws IOException
    */
   private List<String> getPartitionPathsForFullCleaning() {
     // Go to brute force mode of scanning all partitions
-    return FSUtils.getAllPartitionPaths(context, config.getMetadataConfig(), config.getBasePath());
+    try {
+      // Because the partition of BaseTableMetadata has been deleted,
+      // all partition information can only be obtained from FileSystemBackedTableMetadata.
+      FileSystemBackedTableMetadata fsBackedTableMetadata = new FileSystemBackedTableMetadata(context,
```

Comment:
@codope : do we really need to go w/ FileSystemBackedMetadata here? this is causing performance issues for some cases when getPartitionPathsForFullCleaning is being used. 
https://github.com/apache/hudi/issues/6373